### PR TITLE
Fixes for resource docs generator

### PR DESCRIPTION
**Pull request by praneetloke**
This PR contains the following changes/fixes:
* Fix the generation of formal parameters of the function shown in the "Lookup a resource" section by individually generating the formal parameters string.
* "Linkify" the formal parameters in the code snippet of the "Lookup a resource" section.
* Generate the constructor params for Python in code and render the string in the template, so we can control the insertion of word-break tags in the names. This essentially brings it in-sync with how the rest of the constructors were being generated.
* Fix issue with the Go-based Function argument not having the right prefix based on whether or not it was a resource or package-level Function.

The result of these changes can be previewed using the [preview link](http://content-bucket-de682d9.s3-website-us-east-1.amazonaws.com/) provided by Pulumify [here](https://github.com/pulumi/docs/pull/2641#issuecomment-599140931).

### CODE REVIEW

File: pkg/codegen/python/doc.go [hunk 1]
```diff
@@ -91,11 +92,28 @@ func elementTypeToName(el string) string {
...
+// with a key of type `t` and a value of Any.
+func getMapWithTypeName(t string) string {
+	switch t {
+	case "string":
+		return "Dict[str, Any]"
+	case "any":
+		return "Dict[Any, Any]"
+	default:
+		return fmt.Sprintf("Dict[%s, Any]", t)
+	}
```
Comment (justinvp) [inline pkg/codegen/python/doc.go]: I don't think this is correct. Schema.MapType is defined as:

https://github.com/pulumi/pulumi/blob/ee62f38ab998d7aaaefd16742f8b94912a37c57b/pkg/codegen/schema/schema.go#L101-L105

> MapType represents maps from strings to particular element types. 

So the dictionary's key should always be `str` and the value should be what `t` is specifying.

File: pkg/codegen/python/doc.go [hunk 2]
```diff
@@ -63,9 +62,11 @@ func (d DocLanguageHelper) GetLanguageTypeString(pkg *schema.Package, moduleName
 		switch dictionaryTy := t.(type) {
 		case *schema.MapType:
 			elType := dictionaryTy.ElementType.String()
-			return getDictWithTypeName(elementTypeToName(elType))
+			return getMapWithTypeName(elementTypeToName(elType))
 		case *schema.ObjectType:
 			return getDictWithTypeName(tokenToName(dictionaryTy.Token))
+		default:
+			return "Dict[Any, Any]"
```
Comment (justinvp) [inline pkg/codegen/python/doc.go]: Shouldn't this be:

```suggestion
			return "Dict[str, Any]"
```

?

Comment (praneetloke) [reply pkg/codegen/python/doc.go]: Yes, absolutely! Based on your other comment, the key for this should always be of type string.

File: pkg/codegen/python/doc.go (hunk 1 above)
Comment (praneetloke) [reply pkg/codegen/python/doc.go]: Good catch! Fixed it in the latest commit.

**Pull request merged**